src/test/regress/sql/multi_prepare_plsql.sql

```diff
@@ -247,6 +247,104 @@ SELECT plpgsql_test_2();
 -- SELECT plpgsql_test_6(155);
 -- SELECT plpgsql_test_6(1555);
 
+-- test router executor parameterized PL/pgsql functions
+CREATE TABLE temp_table (
+	key int,
+	value int
+);
+SELECT master_create_distributed_table('temp_table','key','hash');
+SELECT master_create_worker_shards('temp_table',4,1);
+
+CREATE OR REPLACE FUNCTION no_parameter_insert() RETURNS void as $$
+BEGIN
+	INSERT INTO temp_table (key) VALUES (0);
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT no_parameter_insert();
+SELECT no_parameter_insert();
+SELECT no_parameter_insert();
+SELECT no_parameter_insert();
+SELECT no_parameter_insert();
+SELECT no_parameter_insert();
+
+CREATE OR REPLACE FUNCTION single_parameter_insert(key_arg int) RETURNS void as $$
+BEGIN
+	INSERT INTO temp_table (key) VALUES (key_arg);
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT single_parameter_insert(1);
+SELECT single_parameter_insert(2);
+SELECT single_parameter_insert(3);
+SELECT single_parameter_insert(4);
+SELECT single_parameter_insert(5);
+SELECT single_parameter_insert(6);
+
+CREATE OR REPLACE FUNCTION double_parameter_insert(key_arg int, value_arg int) RETURNS void as $$
+BEGIN
+	INSERT INTO temp_table (key, value) VALUES (key_arg, value_arg);
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT double_parameter_insert(1, 10);
+SELECT double_parameter_insert(2, 20);
+SELECT double_parameter_insert(3, 30);
+SELECT double_parameter_insert(4, 40);
+SELECT double_parameter_insert(5, 50);
+SELECT double_parameter_insert(6, 60);
+
+-- check inserted values
+SELECT * FROM temp_table ORDER BY key, value;
+
+-- check router executor select
+CREATE OR REPLACE FUNCTION partition_column_select(key_arg int) RETURNS TABLE(key int, value int) AS $$
+DECLARE
+BEGIN
+    RETURN QUERY
+	SELECT
+		temp_table.key,
+		temp_table.value
+	FROM
+		temp_table
+	WHERE
+		temp_table.key = key_arg
+	ORDER BY
+		key,
+		value;
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT partition_column_select(1);
+SELECT partition_column_select(2);
+SELECT partition_column_select(3);
+SELECT partition_column_select(4);
+SELECT partition_column_select(5);
```

Comment:
That's a good point. This was my initial idea, but for failing case, the order of errors changes from execution to execution.

```sql
SELECT partition_column_select(6);
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57637
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57638
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57637
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57638
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57637
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57638
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57637
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57638
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57637
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
WARNING:  there is no parameter $1
CONTEXT:  while executing command on localhost:57638
PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
ERROR:  failed to execute job 780037
DETAIL:  Failure due to failed task 2
CONTEXT:  PL/pgSQL function partition_column_select(integer) line 4 at RETURN QUERY
```